spacy_experimental/coref/pytorch_coref_model.py

```diff
@@ -6,6 +6,36 @@
 EPSILON = 1e-7
 
 
+class MiniClusterer(nn.Module):
+    def __init__(
+        self,
+        dim: int,
+        dist_emb_size: int,
+        hidden_size: int,
+        n_layers: int,
+        dropout: float,
+        rough_k: int,
+        batch_size: int,
+    ):
+        super().__init__()
+        self.dropout = torch.nn.Dropout(dropout)
+        self.batch_size = batch_size
+        self.pairwise = DistancePairwiseEncoder(dist_emb_size, dropout)
+
+        self.word_transform = torch.nn.Linear(in_features=dim, out_features=dim)
+
+        self.rough_scorer = RoughScorer(dim, dropout, rough_k)
+
+    def forward(
+        self, word_features: torch.Tensor, lengths: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        words = self.dropout(self.word_transform(word_features))
+        top_rough_scores, top_indices = self.rough_scorer(words)
+        top_rough_scores = add_dummy(top_rough_scores, eps=True)
+
+        return top_rough_scores, top_indices
+
+
 class CorefClusterer(nn.Module):
```

Comment:
Could we also version this change so that its easier to compare? We can keep the old `CorefClusterer` and the new lighter weight one as `CorefClusterer_v2`.